### PR TITLE
MSI-2162 Add API reference

### PR DESCRIPTION
**Pull request by lorikrell**
## Purpose of this pull request

Add a new API reference. Use content from https://github.com/magento-engcom/msi/wiki/Magento-MSI-APIs.

Merge content into the new page, remove these pages, redirect:
- /guides/v2.3/inventory/catalog-inventory-replacements.html
- /guides/v2.3/inventory/scalable-inventory-replacements.html

MSI issue:
https://github.com/magento-engcom/msi/issues/2162

Cleaned up other pages with variables, links to new content.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/inventory/catalog-inventory-replacements.html
- https://devdocs.magento.com/guides/v2.3/inventory/scalable-inventory-replacements.html

Add new page
https://devdocs.magento.com/guides/v2.3/inventory/inventory-api-reference.html

### CODE REVIEW

File: guides/v2.3/inventory/inventory-api-reference.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+[G`etProductSalableQtyInterface`](https://github.com/magento-engcom/msi/blob/2.3-develop/app/code/Magento/InventorySalesApi/Api/GetProductSalableQtyInterface.php)  | Get product salable quantity (aggregated stock quantity including reservations) |  [`StockStateInterface::getStockQty`](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/CatalogInventory/Api/StockStateInterface.php#L72) or just retrieve `Qty` from [`StockStatus`](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/CatalogInventory/Api/Data/StockStatusInterface.php#L65) entity
```
Comment (keharper) [inline guides/v2.3/inventory/inventory-api-reference.md]:
```suggestion
[`GetProductSalableQtyInterface`](https://github.com/magento-engcom/msi/blob/2.3-develop/app/code/Magento/InventorySalesApi/Api/GetProductSalableQtyInterface.php)  | Get product salable quantity (aggregated stock quantity including reservations) |  [`StockStateInterface::getStockQty`](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/CatalogInventory/Api/StockStateInterface.php#L72) or just retrieve `Qty` from [`StockStatus`](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/CatalogInventory/Api/Data/StockStatusInterface.php#L65) entity
```

File: guides/v2.3/inventory/inventory-api-reference.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+[`SourceSelectionServiceInterface`]  |  Service to run a specified Source Selection Algorithm (SSA) based on provided Inventory Request object
+
+## Inventory Export Aggregated Stock Data API
+
+The following services reside under the [`Magento/InventoryExportStockApi/Api`](https://github.com/magento-engcom/msi/tree/2.3-develop/app/code/Magento/InventoryExportStockApi/Api) namespace.
+
+Service | Description
+--- | ---
+[`ExportStockIndexDataInterface`](https://github.com/magento-engcom/msi/blob/2.3-develop/app/code/Magento/InventoryExportStockApi/Api/ExportStockIndexDataInterface.php)  |  Service to return salable quantities for a provided sales channel not including placed reservations for completed orders. Less precise with high performance and speed.
+[`ExportStockSalableQtyInterface`](https://github.com/magento-engcom/msi/blob/2.3-develop/app/code/Magento/InventoryExportStockApi/Api/ExportStockSalableQtyInterface.php)  |  Service to return salable quantities for a provided sales channel including placed reservations (for completed orders). Preceise with reduced performance and slow speed.
```
Comment (keharper) [inline guides/v2.3/inventory/inventory-api-reference.md]: Typo: Preceise  Can't figure out what it should be.

File: guides/v2.3/inventory/inventory-api-reference.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+- `Status`
+- `Stock` (Helper)
+- `Stock` (model/source)
+- `Stock`
+- `StockFactory`
+
+## {{site.data.var.im}} API
+
+The following services reside under the [`Magento/InventoryApi/Api`](https://github.com/magento-engcom/msi/tree/2.3-develop/app/code/Magento/InventoryApi/Api) namespace, replacing deprecated services and APIs.
+
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply guides/v2.3/inventory/inventory-api-reference.md]: Good point, editing!

File: guides/v2.3/inventory/inventory-api-reference.md (hunk 2 above)
Comment (lorikrell) [reply guides/v2.3/inventory/inventory-api-reference.md]: Revised these two, thanks!

File: guides/v2.3/inventory/reservations.md [hunk 4]
```diff
@@ -18,15 +18,15 @@ Magento creates a reservation for each product when the following events occur:
...
 
 * **StockItem quantity**. The StockItem quantity is the aggregated amount of inventory from all the physical sources for the current sales channel. If the Baltimore source has 20 units of a product, the Austin source has 25 units of the same product, while the Reno source has 10, and all these sources are linked to Stock A, then the StockItem count for thus product is 55 (20 + 15 + 10). (When items are shipped, the Inventory indexer updates the quantities available at each source.)
 
-* **Outstanding reservations**. Magento totals all the initial reservations that have not been compensated. This number will always be negative. If customer A has a reservation for 10 items, and customer B has a reservation 5 for items, then outstanding reservations for the product total -15. 
+* **Outstanding reservations**. Magento totals all the initial reservations that have not been compensated. This number will always be negative. If customer A has a reservation for 10 items, and customer B has a reservation 5 for items, then outstanding reservations for the product total -15.
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply guides/v2.3/inventory/reservations.md]: Hmm yes, you ultimately want the reservations to clear out to 0 after completing orders either through shipments, cancelations, refunds, etc. If they are not 0, then it should be an order with outstanding amounts to fulfill, etc. If it is completed in some way, then they have an issue to use the CLI! Will add some notes.

File: guides/v2.3/inventory/reservations.md [hunk 5]
```diff
@@ -85,19 +85,22 @@ The following example shows the sequence of reservations generated for a simple
...
 
 Often, all initial reservations produced in a single day cannot compensated that same day.  This situation could occur when a customer places an order minutes before the cron job begins or makes the purchase with an offline payment method, such as a bank transfer. The compensated reservation sequences remain in the database until they have all been compensated. This practice does not interfere with reservation calculations, because the total for each reservation is 0.
 
+{: .bs-callout .bs-callout-info }
+{{site.data.var.im}} provides commands to detect and manage reservation inconsistencies. See [Inventory CLI reference](https://devdocs.magento.com/guides/v2.3/inventory/inventory-cli-reference.html).
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply guides/v2.3/inventory/reservations.md]: I can change it, not required.

**Pull request merged**